ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -66,7 +89,34 @@ internal constructor(
           .map { submission -> submission.value().map { it.locationOfInterest } }
           .map { locationOfInterest -> locationOfInterestHelper.getLabel(locationOfInterest) }
       )
+
+    continueResults = continueClicks.toObservable().switchMapSingle { onContinueClicked() }
   }
 
   fun loadSubmissionDetails(args: DataCollectionFragmentArgs) = argsProcessor.onNext(args)
+
+  fun addTaskViewModel(taskViewModel: AbstractTaskViewModel) {
+    taskViewModels.value!!.add(taskViewModel)
+  }
+
+  /**
+   * Validates the user's input and returns an error string if the user input was invalid.
+   * Progresses to the next Data Collection screen if the user input was valid.
+   */
+  fun onContinueClicked(): Single<String> {
+    val currentTask = taskViewModels.value!![currentPosition.value!!]
+    val validationError = currentTask.validate()
+    if (validationError == null) {
+      // TODO(#1146): Handle the scenario when the user clicks next on the last Task. This will
+      //  include persisting the list of responses to the database
+      currentPosition.postValue(currentPosition.value!! + 1)
+
+      responses[currentTask.task.id] = currentTask.response.value?.orElse(null)
+      return Single.never()
+    } else {
+      popups.get().showError(validationError)
+    }
+
+    return Single.just(validationError)
```

Comment:
It seems like we are displaying the error twice:

1. First time when we directly do, `popups.get().showError(validationError)`
2. Second time in `handleContinueClickError()`

Please fix.

Reply:
Done!